### PR TITLE
feat(langchain): implement `UnsupportedContentMiddleware`

### PR DESCRIPTION
**Pull request by ccurme**
Implements `UnsupportedContentMiddleware` which uses [model profiles](https://docs.langchain.com/oss/python/langchain/models#model-profiles) to replace unsupported content (e.g., images for a model that doesn't support vision) with a text notice. This would otherwise brick a langgraph thread once the unsupported message is checkpointed.

Used for this fix in deepagents: https://github.com/langchain-ai/deepagents/pull/6366.

### CODE REVIEW

**Review by chilkotiKartik — COMMENTED** [review]
Nice addition! The middleware cleanly intercepts unsupported blocks before checkpointing in LangGraph.

One small observation in `is_supported`:
```python
if in_tool_message:
    tool_field = _TOOL_MESSAGE_FIELD_BY_BLOCK_TYPE.get(block_type)
    if tool_field is not None and profile.get(tool_field) is False:
        return False
return profile.get(field) is not False
```
If a model's profile specifies `image_tool_message: True` but `image_inputs: False` (or vice versa where tool message support differs from general user input), the method currently continues to `profile.get(field) is not False` and returns `False` because `image_inputs` is False.

If `profile.get(tool_field)` is explicitly boolean (`is not None`), should it take precedence over `profile.get(field)` when `in_tool_message` is True?
e.g.:
```python
if in_tool_message:
    tool_field = _TOOL_MESSAGE_FIELD_BY_BLOCK_TYPE.get(block_type)
    if tool_field is not None and tool_field in profile:
        return bool(profile[tool_field])
```
Aside from this edge case, the implementation and tests look solid!

**Comment by chilkotiKartik** [comment]
LGTM! Reviewed `feat(langchain): implement UnsupportedContentMiddleware`. Gracefully intercepting and filtering unsupported multi-modal content blocks prevents runtime exceptions when routing to text-only model providers.

File: libs/langchain_v1/langchain/agents/middleware/unsupported_content.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        block_type = block["type"]
+        field = _PROFILE_FIELD_BY_BLOCK_TYPE.get(block_type)
+        if field is None:
+            return True
+        if block_type == "file" and (
+            "base64" not in block or block.get("mime_type") != _PDF_MIME_TYPE
+        ):
+            # URL- and file-ID-backed references are provider-managed, and no profile
+            # field describes non-PDF payloads (`.docx`, `.pptx`, ...).
+            return True
```
[the remark this answers is not in the document]
Comment (ccurme) [reply libs/langchain_v1/langchain/agents/middleware/unsupported_content.py:96]: This is a lift-and-shift from deepagents, I don't want to mix in functional changes. We can follow up in a separate PR.

**Comment by ccurme** [comment]
It turns out to make this really useful we need some augmentations that are not yet in model profiles. There are open issues in models.dev proposing them (e.g., enumerating supported mime-types). I'm going to close this and implement the middleware in deepagents for now to avoid hard-coding things in `langchain`: https://github.com/langchain-ai/deepagents/pull/6366
